config/configgrpc/configgrpc.go

```diff
@@ -102,13 +80,13 @@ func GrpcSettingsToDialOptions(settings GRPCSettings) ([]grpc.DialOption, error)
 		}
 	}
 
-	if settings.TLSConfig.CaCert != "" && !settings.TLSConfig.UseSecure {
-		creds, err := credentials.NewClientTLSFromFile(settings.TLSConfig.CaCert, settings.TLSConfig.ServerNameOverride)
+	if settings.TLSConfig.CAFile != "" && settings.TLSConfig.UseInsecure {
+		creds, err := credentials.NewClientTLSFromFile(settings.TLSConfig.CAFile, settings.TLSConfig.ServerName)
 		if err != nil {
 			return nil, err
 		}
 		opts = append(opts, grpc.WithTransportCredentials(creds))
-	} else if settings.TLSConfig.UseSecure {
+	} else if !settings.TLSConfig.UseInsecure {
```

Comment:
I would expect the logic around arguments in `TLSConfig` to be in a function in `TLSConfig` to ensure separation of concerns, I may be wrong but can we try to see if that is possible?

Reply:
Also this code can be simplified (I personally see this more readable):
```golang
if settings.TLSConfig.UseInsecure {
  if settings.TLSConfig.CAFile != "" {

  } else {

  }
} else {

}
```

Reply:
Two good points- I'll clean this up.